src/main/java/com/lambdaschool/oktafoundation/controllers/PartnerApplicationController.java

```diff
@@ -0,0 +1,101 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/partnerapps")
+public class PartnerApplicationController
+{
+    @Autowired
+    private PartnerApplicationService partnerApplicationService;
+
+    @Autowired
+    private UserService userService;
+
+    // Gets all applications w.a.
+    @PreAuthorize("hasAnyRole('ADMIN')")
+    @GetMapping(value = "/all", produces = "application/json")
+    public ResponseEntity<?> listAllPartApplications()
+    {
+        List<PartnerApplication> myPartApps = partnerApplicationService.findAll();
+        return new ResponseEntity<>(myPartApps, HttpStatus.OK);
+    }
+
+    // Gets one partner application by id w.a.
+    @PreAuthorize("hasAnyRole('ADMIN')")
+    @GetMapping(value = "/partapp/{id}", produces = "application/json")
+    public ResponseEntity<?> getPartAppById(@PathVariable long id)
+    {
+        PartnerApplication partnerApplication = partnerApplicationService.findPartAppById(id);
+        return new ResponseEntity<>(partnerApplication, HttpStatus.OK);
+    }
+    // Gets the partner application by id and updates the status to 'accept' w.a.
+    @PatchMapping(value = "/app/{id}/status", consumes = "application/json")
+    public ResponseEntity<?> updateUser(@RequestBody PartnerApplication updatePartApp, @PathVariable long id)
+    {
+        partnerApplicationService.updatePartAppStatus(updatePartApp, id);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    // Users use this endpoint to update their own applications
+    @PatchMapping(value = "/partapp/{id}", consumes = "application/json")
+    public ResponseEntity<?> updateList(Authentication authentication, @RequestBody PartnerApplication updatePartApp, @PathVariable long id)
+    {
+        User user = userService.findByName(authentication.getName());
+        List<PartnerApplication> userList = new ArrayList<>();
+        System.out.println(authentication.getName());
+
+
+
+        if(user.getApplications().size() == 0) throw new ResourceNotFoundException("This user does not have this application with this id " + id);
+
+        for (PartnerApplication p : user.getPartnerApplications())
+        {
+            if(p.getPartappid() == id)
+            {
+                userList.add(p);
+                System.out.println("this is application " + p);
+
+            }
+        }
+
+        partnerApplicationService.update(user, updatePartApp, id);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    // posts new application to database w.a.
+    @PostMapping(value = "/partapp/new", consumes = "application/json")
+    public ResponseEntity<?> addNewPartnerApplication(Authentication authentication, @RequestBody PartnerApplication newapp) throws URISyntaxException
+    {
+        User user = userService.findByName(authentication.getName());
+
+        newapp.setPartappid(0);
+        newapp = partnerApplicationService.saveByAuth(user, newapp);
```

Comment:
The controller is passing two parameters here (which I'm pretty sure is correct), but in the serviceImpl, it's defined to take three parameters?

It looks like you just need to delete that third parameter `long id` in the definition of saveByAuth in ApplicationServiceImpl and PartnerApplicationServiceImpl.

Easy thing to miss! Java has everything so interconnected, if you change something in one place, it's gotta be changed in three others too.

Reply:
fixed!